### PR TITLE
[qt] Add patches to fix missing includes

### LINKED ISSUE #19226: [qt5-base] build failure: qt5-base[core]:x64-linux -> 5.15.2
**Host Environment**
- OS: x64-linux
- Compiler: gcc (GCC) 11.1.1 20210531 (Red Hat 11.1.1-3)

**To Reproduce**
`./vcpkg install --recurse qt5`
```
**Failure logs** 
Computing installation plan...                                                                                                                                  
The following packages will be built and installed:                                                                                                             
    qt5[activeqt,core,declarative,essentials,imageformats,multimedia,networkauth,quickcontrols2,svg,tools]:x64-linux -> 5.15.2                                  
  * qt5-base[core]:x64-linux -> 5.15.2#9                                                                                                                        
  * qt5-declarative[core]:x64-linux -> 5.15.2                                                                                                                   
  * qt5-imageformats[core]:x64-linux -> 5.15.2                                                                                                                  
  * qt5-multimedia[core]:x64-linux -> 5.15.2                                                                                                                    
  * qt5-networkauth[core]:x64-linux -> 5.15.2                                                                                                                   
  * qt5-quickcontrols2[core]:x64-linux -> 5.15.2                                                                                                                
  * qt5-svg[core]:x64-linux -> 5.15.2                                                                                                                           
  * qt5-tools[core]:x64-linux -> 5.15.2                                                                                                                         
Additional packages (*) will be modified to complete this operation.                                                                                            
Detecting compiler hash for triplet x64-linux...                                                                                                                
Could not locate cached archive: /home/chris/.cache/vcpkg/archives/e4/e4541f892fb30bf27178c1c192eaa1edc63f7462ffe70de32498384a9fffb385.zip                      
Could not locate cached archive: /home/chris/.cache/vcpkg/archives/f1/f11b3930664f334266f2e50ee0c23b456e816139d54a4945d6bb349e8f3b9af6.zip                      
Could not locate cached archive: /home/chris/.cache/vcpkg/archives/72/72af6d30698ae4d1cd02649650a5593921fafa0301c7ffa31b10ab93e734910a.zip                      
Could not locate cached archive: /home/chris/.cache/vcpkg/archives/81/81eefde62483ee882b5703956ebe53663aadf910ebe5a52693a565ef61109377.zip                      
Could not locate cached archive: /home/chris/.cache/vcpkg/archives/75/75d0d91c0940aa32c5c9193a7dac5cf4626638e0654a88c359347c50d8bda978.zip                      
Could not locate cached archive: /home/chris/.cache/vcpkg/archives/0c/0c7265d30d76abdc6a56a4e5b2922a58bc7abf846243200850e74b107150494d.zip                      
Could not locate cached archive: /home/chris/.cache/vcpkg/archives/d8/d8740e96b4b9ab5ece4568cf132b109d2accde0135e0884531cb4fa204bbcd0f.zip                      
Could not locate cached archive: /home/chris/.cache/vcpkg/archives/52/52eef8e14c9d8a9cfed3913a935b54246e5d18ce22845e5baaeb56c3bbcfa528.zip                      
Could not locate cached archive: /home/chris/.cache/vcpkg/archives/d4/d4510a74d349592a8d7565faa4c679b4802cba954ab16196616d4d6a77b9939f.zip                      
Starting package 1/9: qt5-base:x64-linux                                                                                                                        
Building package qt5-base[core]:x64-linux...                                                                                                                    
CMake Warning at ports/qt5-base/portfile.cmake:11 (message):                                                                                                    
  qt5-base currently requires some packages from the system package manager,                                                                                    
  see https://doc.qt.io/qt-5/linux-requirements.html                                                                                                            
Call Stack (most recent call first):                                                                                                                            
  scripts/ports.cmake:141 (include)                                                                                                                             
                                                                                                                                                                
                                                                                                                                                                
CMake Warning at ports/qt5-base/portfile.cmake:12 (message):                                                                                                    
  qt5-base for qt5-x11extras requires several libraries from the system
  package manager.  Please refer to

    https://github.com/microsoft/vcpkg/blob/master/scripts/azure-pipelines/linux/provision-image.sh
    for a complete list of them.

Call Stack (most recent call first):
  scripts/ports.cmake:141 (include)


-- Figuring out qt target mkspec. Target arch x64
-- Target mkspec set to: linux-g++
-- Host mkspec not set. Qt's own buildsystem will try to figure out the host system
-- Using /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/downloads/qtbase-everywhere-src-5.15.2.tar.xz
-- Cleaning sources at /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/buildtrees/qt5-base/src/5.15.2-70178539f6.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/downloads/qtbase-everywhere-src-5.15.2.tar.xz
-- Applying patch patches/winmain_pro.patch
-- Applying patch patches/windows_prf.patch
-- Applying patch patches/qt_app.patch
-- Applying patch patches/gui_configure.patch
-- Applying patch patches/icu.patch
-- Applying patch patches/xlib.patch
-- Applying patch patches/egl.patch
-- Applying patch patches/zstdd.patch
-- Applying patch patches/mysql_plugin_include.patch
-- Applying patch patches/mysql-configure.patch
-- Applying patch patches/Qt5BasicConfig.patch
-- Applying patch patches/Qt5PluginTarget.patch
-- Applying patch patches/create_cmake.patch
-- Applying patch patches/Qt5GuiConfigExtras.patch
-- Using source at /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/buildtrees/qt5-base/src/5.15.2-70178539f6.clean
-- Configuring x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
    Command failed: /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/buildtrees/qt5-base/src/5.15.2-70178539f6.clean/configure -confirm-license -opensource -verbose -system-zlib -system-libjpeg -system-libpng -system-freetype -system-pcre -system-doubleconversion -system-sqlite -system-harfbuzz -icu -no-vulkan -no-angle -no-glib -openssl-linked -no-sql-psql -no-sql-mysql --vulkan=no -fontconfig -xcb-xlib -xcb -linuxfb -nomake examples -nomake tests -static LIBJPEG_LIBS=/home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libjpeg.a ZLIB_LIBS=/home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libz.a "LIBPNG_LIBS=/home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libpng16d.a /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libz.a" PCRE2_LIBS=/home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libpcre2-16.a "FREETYPE_LIBS=/home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libfreetyped.a /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libbz2d.a /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libpng16d.a /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libz.a /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libbrotlidec-static.a /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libbrotlicommon-static.a" "ICU_LIBS=/home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libicui18n.a /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libicutu.a  /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libicuuc.a /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libicuio.a /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libicudata.a" QMAKE_LIBS_PRIVATE+=/home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libbz2d.a QMAKE_LIBS_PRIVATE+=/home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libpng16d.a "QMAKE_LIBS_PRIVATE+=/home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libicui18n.a /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libicutu.a  /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libicuuc.a /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libicuio.a /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libicudata.a" QMAKE_LIBS_PRIVATE+=/home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libzstdd.a "SQLITE_LIBS=/home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libsqlite3.a -ldl -lpthread" "HARFBUZZ_LIBS=-lharfbuzz -lm -lfreetyped -lbz2d -lpng16d -lm -lz -lbrotlidec-static -lbrotlicommon-static" "OPENSSL_LIBS=/home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libssl.a /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libcrypto.a -ldl -lpthread" "FONTCONFIG_LIBS=/home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libfontconfig.a /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libfreetyped.a /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/libexpat.a -luuid" -debug -prefix /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux -hostprefix /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/tools/qt5/debug -hostlibdir /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/tools/qt5/debug/lib -hostbindir /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/tools/qt5/debug/bin -archdatadir /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/tools/qt5/debug -datadir /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/share/qt5/debug -plugindir /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/plugins -qmldir /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/qml -headerdir /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/include/qt5 -libexecdir /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/tools/qt5/debug -bindir /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/bin -libdir /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib -I /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/include -I /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/include/qt5 -L /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib -L /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/installed/x64-linux/debug/lib/manual-link -platform linux-g++
    Working Directory: /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/buildtrees/qt5-base/x64-linux-dbg
    Error code: 2
    See logs for more information:
      /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/buildtrees/qt5-base/config-x64-linux-dbg-out.log
      /home/chris/FreeCAD/workspace/FreeCAD/freecad-source/tools/vcpkg/buildtrees/qt5-base/config-x64-linux-dbg-err.log

Call Stack (most recent call first):
  ports/qt5-base/cmake/configure_qt.cmake:127 (vcpkg_execute_required_process)
  ports/qt5-base/portfile.cmake:363 (configure_qt)
  scripts/ports.cmake:141 (include)
```
[config-x64-linux-dbg-err.log](https://github.com/microsoft/vcpkg/files/6899938/config-x64-linux-dbg-err.log)
[config-x64-linux-dbg-out.log](https://github.com/microsoft/vcpkg/files/6899940/config-x64-linux-dbg-out.log)

**Additional context**

Initial failed due to autopoint not being installed so had to run `sudo dnf install gettext-devel`

### PR DESCRIPTION
**Pull request by nicolasfella**
This is needed to build with GCC11

Fixes #19226

### CODE REVIEW

**Comment by JackBoosY** [comment]
cc @Neumann-A for review this PR.

File: ports/qt5-declarative/portfile.cmake [hunk 1]
```diff
@@ -1,2 +1,2 @@
 include(${CURRENT_INSTALLED_DIR}/share/qt5/qt_port_functions.cmake)
-qt_submodule_installation()
\ No newline at end of file
+qt_submodule_installation(PATCHES limits_include.patch)
```
Comment (JackBoosY) [inline ports/qt5-declarative/portfile.cmake:2]: Please update qt5-declarative's port version.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by Neumann-A — APPROVED** [review]

**Comment by Be-ing** [comment]
Ready for merge?

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for fixing GCC 11 :)

**Pull request merged**